diff --git a/src/vs/base/browser/ui/menu/menu.ts b/src/vs/base/browser/ui/menu/menu.ts
index 402ba66..ff37ece 100644
--- a/src/vs/base/browser/ui/menu/menu.ts
+++ b/src/vs/base/browser/ui/menu/menu.ts
@@ -900,11 +900,11 @@ class SubmenuMenuActionViewItem extends BaseMenuActionViewItem {
 			// This allows the menu constructor to calculate the proper max height
 			const computedStyles = getWindow(this.parentData.parent.domNode).getComputedStyle(this.parentData.parent.domNode);
 			const paddingTop = parseFloat(computedStyles.paddingTop || '0') || 0;
-			this.submenuContainer.style.position = 'fixed';
-			this.submenuContainer.style.top = '0';
-			this.submenuContainer.style.left = '0';
-			// Fix to #263546, for submenu of treeView view/item/context z-index issue - ensure submenu appears above other elements
-			this.submenuContainer.style.zIndex = '1';
+		this.submenuContainer.style.position = 'fixed';
+		this.submenuContainer.style.top = '0';
+		this.submenuContainer.style.left = '0';
+		// Fix to #263546, for submenu of treeView view/item/context z-index issue - ensure submenu appears above other elements
+		this.submenuContainer.style.zIndex = '3500';
 
 			this.parentData.submenu = new Menu(this.submenuContainer, this.submenuActions.length ? this.submenuActions : [new EmptySubmenuAction()], this.submenuOptions, this.menuStyle);
 
diff --git a/src/vs/base/browser/ui/menu/menubar.css b/src/vs/base/browser/ui/menu/menubar.css
index 17fbe89..23a4afc 100644
--- a/src/vs/base/browser/ui/menu/menubar.css
+++ b/src/vs/base/browser/ui/menu/menubar.css
@@ -60,7 +60,7 @@
 	position: fixed;
 	left: 0px;
 	opacity: 1;
-	z-index: 2000;
+	z-index: 3500;
 }
 
 .menubar.compact .menubar-menu-items-holder {

